convert2rhel/utils.py

```diff
@@ -114,7 +114,7 @@ def store_content_to_file(filename, content):
 def restart_system():
     from convert2rhel.toolopts import tool_opts
     if tool_opts.restart:
-        run_subprocess("reboot")
+        run_subprocess("shutdown -r now")
```

Comment:
I haven't tested that but I'd say this change is going to have the same outcome. Both `reboot` and `shutdown` are located in `/sbin/` on RHEL 6-based systems, and both are installed through the `upstart` package. My wild guess is that `/sbin/` is not in paths where the python subprocess library looks for executables.

I needed to workaround probably the same type of issue at https://github.com/oamg/convert2rhel/blob/v0.20/convert2rhel/pkghandler.py#L736.

I quickly checked CentOS Linux 6/7/8 now and `/sbin/reboot` exists on all.

```suggestion
        run_subprocess("/sbin/reboot")
```

Reply:
To follow up, it could be something different, because I have one machine where only shutdown command exists. Also there are vast number of discussions on stack overflow where shutdown -r is recommended over the reboot due to the exactly the reason of possibility of missing the reboot command.

Reply:
Let me check first why shutdown was present and reboot not, perhaps fall back to your proposal

Reply:
ok, no success on my side. I asked George to share the output https://github.com/oamg/convert2rhel/issues/241#issuecomment-818515520

Reply:
Ok, feel free to use shutdown. I did not expect some machine to not have reboot. In that case, we might need to update the documentation as well https://access.redhat.com/articles/2360841.

Out of curiosity, can you check what package the shutdown comes from on the machine where reboot is not available (`rpm -qf /sbin/shutdown`)?

Reply:
[root@d84744d80e20 /]# rpm -qf /sbin/shutdown
error: file /sbin/shutdown: No such file or directory
[root@d84744d80e20 /]# rpm -qf /sbin/reboot  
error: file /sbin/reboot: No such file or directory

But shutdown is available:)

Reply:
anyway, let's wait for George. I have no idea how env={'LC_ALL': 'C'} affects the existence of reboot

Reply:
In that case you can run `which shutdown` to get the path of the executable and then `rpm -qf <path>`.
BTW, what's the machine on which you're experiencing this?

Reply:
> anyway, let's wait for George. I have no idea how env={'LC_ALL': 'C'} affects the existence of reboot

@ZhukovGreen  becuase it drops all existing evironment variables. It's the same case as I already reported somewhere. corerctly it should get  all existing variables and then set `LC_ALL` in the dict so the system setup is propagated. When the `$PATH` envar is dropped, the function search just in /bin /usr/bin directories. It ignores /sbin and /usr/sbin - which are usually set via $PATH explicitly on the system.

Reply:
> [root@d84744d80e20 /]# rpm -qf /sbin/shutdown
> error: file /sbin/shutdown: No such file or directory
> [root@d84744d80e20 /]# rpm -qf /sbin/reboot
> error: file /sbin/reboot: No such file or directory
> 
> But shutdown is available:)

You should search /usr/sbin/shutdown in such a case

Reply:
@pirat89, but ` /usr/sbin/shutdown` in N/A on CentOS/OL Linux 6. In my testing, ` /sbin/shutdown` is available in all CentOS/OL Linux 6/7/8. And in CentOS/OL Linux 7+ it's a symlink to the ` /usr/sbin/shutdown`. But I wondered what package owns the @ZhukovGreen's shutdown executable. I'd like to find out why some package creates the ` /sbin/shutdown` symlink (on my testing machines) and some not (on Artem's testing machines).

Reply:
@bocekm you are right, on EL/OL 6 it's `/sbin/shutdown/ as there is different FHS.